geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/KryoLazyDensityCoprocessor.scala

```diff
@@ -69,85 +69,115 @@ class KryoLazyDensityCoprocessor extends KryoLazyDensityService with Coprocessor
+          }
+          results.clear()
+        } while (hasMore)
+      })
 
       val result: Array[Byte] = KryoLazyDensityUtils.encodeResult(densityResult)
       response = DensityResponse.newBuilder.setSf(ByteString.copyFrom(result)).build
     } catch {
-      case ioe: IOException => {
+      case ioe: IOException =>
         ResponseConverter.setControllerException(controller, ioe)
-      }
-      case cnfe: ClassNotFoundException => {
-        logger.error(Throwables.getStackTraceAsString(cnfe))
-      }
-      case dse: DeserializationException => {
-        logger.error(Throwables.getStackTraceAsString(dse))
-      }
-    } finally IOUtils.closeQuietly(scanner)
+      case cnfe: ClassNotFoundException =>
+        throw cnfe
+      case dse: DeserializationException =>
+        throw dse
+    } finally CloseQuietly(scanner)
+
     done.run(response)
   }
 }
 
-object KryoLazyDensityCoprocessor extends KryoLazyDensityUtils{
-
-  final private val logger = LoggerFactory.getLogger(classOf[KryoLazyDensityCoprocessor])
+object KryoLazyDensityCoprocessor extends KryoLazyDensityUtils {
 
   private val SFT_OPT = "sft"
+  private val FILTER_OPT = "filter"
+  private val RANGES_OPT = "ranges"
 
   /**
     * Creates an iterator config for the kryo density iterator
     */
   def configure(sft: SimpleFeatureType,
-                filter: Option[Filter],
+                ranges: Seq[Scan],
+                filterList: FilterList,
                 hints: Hints): Map[String, String] = {
     import org.locationtech.geomesa.index.conf.QueryHints.RichHints
     val envelope = hints.getDensityEnvelope.get
     val (width, height) = hints.getDensityBounds.get
     val weight = hints.getDensityWeight
-    configure(sft, filter, envelope, width, height, weight)
+    configure(sft, ranges, filterList, envelope, width, height, weight)
   }
 
+
   protected def configure(sft: SimpleFeatureType,
-                                     filter: Option[Filter],
-                                     envelope: Envelope,
-                                     gridWidth: Int,
-                                     gridHeight: Int,
-                                     weightAttribute: Option[String]): Map[String, String] = {
+                          ranges: Seq[Scan],
+                          filterList: FilterList,
+                          envelope: Envelope,
+                          gridWidth: Int,
+                          gridHeight: Int,
+                          weightAttribute: Option[String]): Map[String, String] = {
     val is = mutable.Map.empty[String, String]
+    var scanRanges : List[String] = List[String]()
+
     is.put(ENVELOPE_OPT, s"${envelope.getMinX},${envelope.getMaxX},${envelope.getMinY},${envelope.getMaxY}")
     is.put(GRID_OPT, s"$gridWidth,$gridHeight")
     weightAttribute.foreach(is.put(WEIGHT_OPT, _))
     is.put(SFT_OPT, SimpleFeatureTypes.encodeType(sft, false))
+
+    ranges.foreach(range =>
```

Comment:
for performance, let's use a StringBuilder here instead of building up a new list and then calling `.mkString` on it